Client/Application/QuickActions.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  QuickActions.swift
+//  Client
+//
+//  Created by Emily Toop on 11/20/15.
+//  Copyright © 2015 Mozilla. All rights reserved.
+//
+
+import Foundation
+
+@available(iOS 9, *)
+enum ShortcutType: String {
+    case NewTab
+    case NewPrivateTab
+
+    init?(fullType: String) {
+        guard let last = fullType.componentsSeparatedByString(".").last else { return nil }
+
+        self.init(rawValue: last)
+    }
+
+    var type: String {
+        return NSBundle.mainBundle().bundleIdentifier! + ".\(self.rawValue)"
+    }
+}
+
+@available(iOS 9, *)
+protocol QuickActionHandlerDelegate {
+    func handleShortCutItemType(type: ShortcutType, userData: [String: NSSecureCoding]?)
+}
+
+@available(iOS 9, *)
+struct QuickActions {
+
+    var launchedShortcutItem: UIApplicationShortcutItem?
+
+    func handleShortCutItem(shortcutItem: UIApplicationShortcutItem, completionBlock: (type: ShortcutType, userData: [String: NSSecureCoding]?)->Void ) -> Bool {
+
+        // Verify that the provided `shortcutItem`'s `type` is one handled by the application.
+        guard let shortCutType = ShortcutType(fullType: shortcutItem.type) else { return false }
+
+        completionBlock(type: shortCutType, userData: shortcutItem.userInfo)
+
+        // Construct an alert using the details of the shortcut used to open the application.
+        print("Shortcut Handled: \(shortcutItem.localizedTitle)")
```

Comment:
Can we use an instance of XCLogger from Loggers.swift instead of print so we can write the log to disk? I think Loggers.browser should work.